public/app/features/alerting/components/AlertingQueryPreview.tsx

```diff
@@ -49,59 +50,66 @@ export const AlertingQueryPreview: FC<Props> = ({ getInstances, onRunQueries, on
         })}
       </TabsBar>
       <TabContent className={styles.tabContent}>
-        {data && data.state === 'Error' ? (
-          <div className={styles.noQueries}>
-            <h4 className={styles.noQueriesHeader}>There was an error :(</h4>
-            <div>{data.error?.data?.error}</div>
-          </div>
-        ) : queries && queries.length > 0 ? (
-          <AutoSizer style={{ width: '100%', height: '100%' }}>
-            {({ width, height }) => {
-              switch (activeTab) {
-                case Tabs.Instances:
-                  return (
-                    <PreviewInstancesTab
-                      isTested={instances.length > 0}
-                      instances={instances}
-                      styles={styles}
-                      width={width}
-                      height={height}
-                      onTest={onTest}
-                    />
-                  );
-
-                case Tabs.Query:
-                default:
-                  if (data) {
-                    return <PreviewQueryTab data={data} width={width} height={height} />;
-                  }
-                  return (
-                    <div className={styles.noQueries}>
-                      <h4 className={styles.noQueriesHeader}>Run queries to view data.</h4>
-                      <Button onClick={onRunQueries}>Run queries</Button>
-                    </div>
-                  );
-              }
-            }}
-          </AutoSizer>
-        ) : (
-          <div className={styles.noQueries}>
-            <h4 className={styles.noQueriesHeader}>No queries added.</h4>
-            <div>Start adding queries to this alert and a visualisation for your queries will appear here.</div>
-            <div>
-              Learn more about how to create alert definitions <Icon name="external-link-alt" />
-            </div>
-          </div>
-        )}
+        {data &&
+          (data.state === 'Error' ? (
+            <EmptyState title="There was an error :(">
+              <div>{data.error?.data?.error}</div>
+            </EmptyState>
+          ) : (
+            <QueriesAndInstances
+              instances={instances}
+              onTest={onTest}
+              data={data}
+              activeTab={activeTab}
+              onRunQueries={onRunQueries}
+              queries={queries}
+            />
+          ))}
       </TabContent>
     </div>
   );
 };
 
+interface PreviewProps {
+  queries: DataQuery[];
+  instances: DataFrame[];
+  onTest: () => void;
+  data: PanelData;
+  activeTab: string;
+  onRunQueries: () => void;
+}
+
+const QueriesAndInstances: FC<PreviewProps> = ({ queries, instances, onTest, data, activeTab, onRunQueries }) => {
+  if (queries && queries.length === 0) {
```

Comment:
Nit: can be rewritten as:
```suggestion
  if (queries?.length === 0) {
```